src/app/features/location/gps.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from "@angular/core";
+import { GeoService } from "./geo.service";
+
+@Injectable({
+  providedIn: "root",
+})
+export class GpsService {
+  constructor(private readonly geoService: GeoService) {}
+  location: { lat: number; lon: number } | null = null;
+
+  async getGpsLocationCoordinates(): Promise<{
+    latitude: number;
+    longitude: number;
+    accuracy: number;
+  }> {
+    if (!("geolocation" in navigator)) {
+      return;
+    }
+
+    const permissionStatus = await navigator.permissions.query({
+      // eslint-disable-next-line prettier/prettier
+      name: "geolocation", //NOSONAR geolocation is necessary
```

Comment:
why is disabling prettier necessary here?

Reply:
I had to disable Prettier for this specific line because it was causing issues with the SonarCloud CI checks. The problem was that Prettier's automatic indentation adjustments were moving the `// NOSONAR geolocation is necessary` comment to the next line, which led to the SonarCloud analysis failing. By disabling Prettier on this line, I was able to ensure that both Prettier and SonarCloud CI checks passed successfully without any conflicts.

Reply:
ah, I see. I think we, as admins, can review and confirm SonarCloud's security hotspot warnings on their platform also. Can you remove these comments once and let me check that?

Reply:
I have marked this as reviewed and safe now :+1: